libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5169,6 +5169,30 @@ MAV_RESULT GCS_MAVLINK::handle_do_set_safety_switch_state(const mavlink_command_
     }
 }
 
+MAV_RESULT GCS_MAVLINK::handle_do_set_sys_cmp_id(const mavlink_command_int_t &packet, const mavlink_message_t &msg)
+{
+    const uint8_t new_system_id = packet.param1;
+    const uint8_t new_component_id = packet.param2;
+    const uint8_t do_reboot = packet.param3;
+
+    if (new_system_id == 0) {
+        return MAV_RESULT_FAILED;
+    }
+
+    if (!AP_Param::set_and_save_by_name("SYSID_THISMAV", new_system_id)) {
```

Comment:
Should check we're not setting the same value into the parameter that it already has here.

Unless it is not configured in storage, in which case we may or may not want to save it....